rxandroidble/src/main/java/com/polidea/rxandroidble2/scan/ScanFilter.java

```diff
@@ -258,29 +258,26 @@ public int getManufacturerId() {
      * Check if the scan filter matches a {@code scanResult}. A scan result is considered as a match
      * if it matches all the field filters.
      */
-    public boolean matches(RxBleInternalScanResult scanResult) {
+    public boolean matches(ScanResultInterface scanResult) {
         if (scanResult == null) {
             return false;
         }
-        BluetoothDevice device = scanResult.getBluetoothDevice();
+        String address = scanResult.getAddress();
         // Device match.
-        if (mDeviceAddress != null
-                && (device == null || !mDeviceAddress.equals(device.getAddress()))) {
+        if (mDeviceAddress != null && !mDeviceAddress.equals(address)) {
             return false;
         }
 
         ScanRecord scanRecord = scanResult.getScanRecord();
 
-        // Scan record is null but there exist filters on it.
-        if (scanRecord == null
-                && (mDeviceName != null || mServiceUuid != null || mManufacturerData != null
-                || mServiceData != null)) {
-            return false;
+        // if scan record is null we cannot continue. For filter to pass, remaining filter properties must be empty
```

Comment:
This actually may not be true as `scanRecord` does not need to be set to match agains the peripheral's name (which can be held in `scanResult`)

Reply:
This is fixed and tested now